### PR TITLE
feat: support dashes in username

### LINKED ISSUE #17: issue with community.routeros.command and community.routeros.facts when there is a dash in the username
##### SUMMARY

If the username contains a dash "-" (something like "ssh-admin"), the connection timeout error will appear.
##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```paste below
2.9.16 and 2.10.2
```

##### CONFIGURATION

```paste below
DEFAULT_ASK_PASS(/home/phil/ansible/ansible.cfg) = False
DEFAULT_HOST_LIST(/home/phil/ansible/ansible.cfg) = ['/home/phil/ansible/myhosts']
DEPRECATION_WARNINGS(/home/phil/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT

Debian buster

##### STEPS TO REPRODUCE

Make the username with a dash inside it (eg ssh-admin) for login on routeros device. And then try to start playbook with him.

```yaml

- name: RouterOS test with network_cli connection
  hosts: mikrotik_server
  gather_facts: false
  tasks:

   - community.routeros.command:
      commands:
        - /system resource print
    register: system_resource_print
  - debug:
      var: system_resource_print.stdout_lines

 - community.routeros.facts:
  - debug:
      msg: "First IP address: {{ ansible_net_all_ipv4_addresses[0] }}"
```

##### EXPECTED RESULTS

PLAY [RouterOS test with network_cli connection] ************************************************************************************************************

TASK [community.routeros.command] ***************************************************************************************************************************
ok: [172.16.10.172]

TASK [debug] ************************************************************************************************************************************************
ok: [172.16.10.172] => {
    "system_resource_print.stdout_lines": [
        [
            "uptime: 1w3d23h36m54s",
            "                  version: 6.46.8 (long-term)",
            "               build-time: Oct/29/2020 08:29:55",
            "              free-memory: 189.3MiB",
            "             total-memory: 224.0MiB",
            "                      cpu: Intel(R)",
            "                cpu-count: 4",
            "            cpu-frequency: 2400MHz",
            "                 cpu-load: 0%",
            "           free-hdd-space: 31.3MiB",
            "          total-hdd-space: 63.5MiB",
            "  write-sect-since-reboot: 3600",
            "         write-sect-total: 3601",
            "        architecture-name: x86_64",
            "               board-name: CHR",
            "                 platform: MikroTik"
        ]
    ]
}

TASK [community.routeros.facts] *****************************************************************************************************************************
ok: [172.16.10.172]

TASK [debug] ************************************************************************************************************************************************
ok: [172.16.10.172] => {
    "msg": "First IP address: 172.16.10.172"
}

PLAY RECAP **************************************************************************************************************************************************
172.16.10.172              : ok=4    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

##### ACTUAL RESULTS

The full traceback is:
  File "/tmp/ansible_community.routeros.command_payload_jqwdps2f/ansible_community.routeros.command_payload.zip/ansible_collections/community/routeros/plugins/module_utils/routeros.py", line 74, in get_capabilities
    capabilities = Connection(module._socket_path).get_capabilities()
  File "/tmp/ansible_community.routeros.command_payload_jqwdps2f/ansible_community.routeros.command_payload.zip/ansible/module_utils/connection.py", line 195, in __rpc__
    raise ConnectionError(to_text(msg, errors='surrogate_then_replace'), code=code)
fatal: [172.16.10.172]: FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "changed": false,
    "invocation": {
        "module_args": {
            "commands": [
                "/system resource print"
            ],
            "interval": 1,
            "match": "all",
            "retries": 10,
            "wait_for": null
        }
    },
    "msg": "timeout value 30 seconds reached while trying to send command: b'/system resource print'"
}

```paste below

```

### PR DESCRIPTION
**Pull request by heuels**
##### SUMMARY

Fixes #17.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

`community.routeros.command`, `community.routeros.facts`

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: changelogs/fragments/18-support-dashes-in-username.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - community.routeros.command - added support for a dash (`-`) in username (https://github.com/ansible-collections/community.routeros/pull/18).
+  - community.routeros.facts - added support for a dash (`-`) in username (https://github.com/ansible-collections/community.routeros/pull/18).
```
Comment (felixfontein) [inline changelogs/fragments/18-support-dashes-in-username.yml:3]:
```suggestion
  - community.routeros.command - added support for a dash (``-``) in username (https://github.com/ansible-collections/community.routeros/pull/18).
  - community.routeros.facts - added support for a dash (``-``) in username (https://github.com/ansible-collections/community.routeros/pull/18).
```

Comment (heuels) [reply changelogs/fragments/18-support-dashes-in-username.yml]: Thanks!

File: changelogs/fragments/18-support-dashes-in-username.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - community.routeros.command - added support for a dash (``-``) in username (https://github.com/ansible-collections/community.routeros/pull/18).
+  - community.routeros.facts - added support for a dash (``-``) in username (https://github.com/ansible-collections/community.routeros/pull/18).
```
Comment (felixfontein) [inline changelogs/fragments/18-support-dashes-in-username.yml:3]: Sorry I missed that in the first run; we're always using the short form in the changelog so far, so let's keep it for consistency:
```suggestion
  - command - added support for a dash (``-``) in username (https://github.com/ansible-collections/community.routeros/pull/18).
  - facts - added support for a dash (``-``) in username (https://github.com/ansible-collections/community.routeros/pull/18).
```

Comment (heuels) [reply changelogs/fragments/18-support-dashes-in-username.yml:3]: Sure!

**Review by felixfontein — APPROVED** [review]

**Comment by heuels** [comment]
@felixfontein, thanks for reviewing!

**Comment by felixfontein** [comment]
@heuels should we do another release (1.0.2), or a wait a bit? The next Ansible release should be on January 5th, so we should definitely release before that, but we can also wait until January 4th :)

**Comment by heuels** [comment]
Let's wait a bit — I just remembered I have #8 to fix. I'll be able to do it this week and after that I guess nothing is stopping us from releasing 1.0.2 🙂

**Comment by felixfontein** [comment]
@heuels would be 1.1.0 though since this is a feature PR, not a bugfix PR

**Comment by felixfontein** [comment]
@heuels any news on #8? Otherwise I'll make a 1.1.0 release with what we have now tomorrow.

**Comment by heuels** [comment]
@felixfontein, sorry, I'm missing all the deadlines 😞

**Pull request merged**